plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergSystemTables.java

```diff
@@ -129,7 +129,7 @@ public void testManifestsTable()
                         "('added_data_files_count', 'integer', '', '')," +
                         "('existing_data_files_count', 'integer', '', '')," +
                         "('deleted_data_files_count', 'integer', '', '')," +
-                        "('partitions', 'array(row(contains_null boolean, lower_bound varchar, upper_bound varchar))', '', '')");
+                        "('partitions', 'array(row(contains_null boolean, contains_nan boolean, lower_bound varchar, upper_bound varchar))', '', '')");
```

Comment:
there seem to be no (changed) test that verifies a value is read from the contains_nan field

Reply:
I added this query
```
        assertQuery(
                "SELECT partition_metadata.* FROM test_schema.\"test_table$manifests\" m " +
                        "CROSS JOIN UNNEST(m.partitions) as partition_metadata(contains_null, contains_nan, lower_bound, upper_bound)",
                "VALUES " +
                        "    (false, false, '2019-09-08', '2019-09-09')," +
                        "    (false, false, '2019-09-09', '2019-09-10')");
```

I've tried without success to come up with a solution for specifying `VALUES ARRAY[ROW(...)]` but didn't find the correct way how to do it so i've fallen back to `UNNEST`

Reply:
I tried 

```
        assertQuery("SELECT partitions FROM test_schema.\"test_table$manifests\"",
                "VALUES ARRAY[ROW(false, false, '2019-09-08', '2019-09-09')], ARRAY[ROW(false, false, '2019-09-09', '2019-09-10')]");
```
which creates on H2 side an ArrayList of `Object[]`

https://github.com/trinodb/trino/blob/master/testing/trino-testing/src/main/java/io/trino/testing/H2QueryRunner.java#L382-L389

but the actual query runner creates an array list of `MaterializedRow` reason why the assertion fails.

Reply:
Cool.
Do you have any tips on how to format this large sql sausage?